for_dict_challenges_bonus.py

```diff
@@ -1,70 +1,90 @@
-5. Вывести идентификаторы сообщений, который стали началом для самых длинных тредов (цепочек ответов).
-
-Весь код стоит разбить на логические части с помощью функций.
-"""
-import random
-import uuid
-import datetime
-
-import lorem
-
-
-def generate_chat_history():
-    messages_amount = random.randint(200, 1000)
-    users_ids = list(
-        {random.randint(1, 10000) for _ in range(random.randint(5, 20))}
-    )
-    sent_at = datetime.datetime.now() - datetime.timedelta(days=100)
-    messages = []
-    for _ in range(messages_amount):
-        sent_at += datetime.timedelta(minutes=random.randint(0, 240))
-        messages.append({
-            "id": uuid.uuid4(),
-            "sent_at": sent_at,
-            "sent_by": random.choice(users_ids),
-            "reply_for": random.choice(
-                [
-                    None,
-                    (
-                        random.choice([m["id"] for m in messages])
-                        if messages else None
-                    ),
-                ],
-            ),
-            "seen_by": random.sample(users_ids,
-                                     random.randint(1, len(users_ids))),
-            "text": lorem.sentence(),
-        })
-    return messages
-
-
-if __name__ == "__main__":
-    print(generate_chat_history())
+#вроде сделала 4 первые пункта (как же это было тяжело :((( ))))
+#пятый пункт в процессе, сдам его потом
+
+
+import random
+import uuid
+import datetime
+
+import lorem
+
+
+def generate_chat_history():
+    messages_amount = random.randint(200, 1000)
+    users_ids = list(
+        {random.randint(1, 10000) for _ in range(random.randint(5, 20))}
+    )
+    sent_at = datetime.datetime.now() - datetime.timedelta(days=100)
+    messages = []
+    for _ in range(messages_amount):
+        sent_at += datetime.timedelta(minutes=random.randint(0, 240))
+        messages.append({
+            "id": uuid.uuid4(),
+            "sent_at": sent_at,
+            "sent_by": random.choice(users_ids),
+            "reply_for": random.choice(
+                [
+                    None,
+                    (
+                        random.choice([m["id"] for m in messages])
+                        if messages else None
+                    ),
+                ],
+            ),
+            "seen_by": random.sample(users_ids, random.randint(1, len(users_ids))),
+            "text": lorem.sentence(),
+        })
+    return messages
+
+
+def more_messages(users_list):
+    ids = [user["sent_by"] for user in users_list]
+    return max(set(ids), key=ids.count)
+
+def more_answers(users_list):
+    ids_list = {}
+    all_reply = [message["reply_for"] for message in users_list]
+    for message in users_list:
+        for reply in all_reply:
+            if reply == message["id"]:
+                ids_list[message["sent_by"]]= ids_list.get(message['sent_by'], 0) + 1
+    return max(ids_list, key=ids_list.get)
+
+def more_views(messages_list):

```

Comment:
Думаю более понятное название
```suggestion
def most_seen_users(messages_list):
```